portal/src/main/webapp/js/src/crosscancer.js

```diff
@@ -645,19 +735,36 @@
 
                                 var stacked = $("#histogram-show-colors").is(":checked");
                                 var outX = width + 1000;
+                               
+                               
+                               var maxYAxis = 0;
+                               if($("#yAxis").val() === "Frequency"){
+                                   for(var i = 0;i < histDataOrg.length;i++){
+                                            if(calculateFrequency(histDataOrg[i], 0, "all") + .05 > maxYAxis){
+                                                maxYAxis = calculateFrequency(histDataOrg[i], 0, "all") + .05;
+                                            }
+                                    }
+                                    maxYAxis = Math.min(maxYAxis, 1.0);
+                                    $("#sliderLabel").text("Min. alteration ");
+                                    $("#maxLabel").text(Math.ceil(100*maxYAxis)+"%");
+                                    $("#sliderEventTest").slider( "option", "max", Math.ceil(100*maxYAxis) );
+                                    $("#currentValue").text(sliderValue+"%");
```

Comment:
this could maybe be set at slider onchange event? I would advocate that everything that needs to change when slider is changed, is changed at onchange. Keeps code clear.